### ISSUE TITLE
Content type text/other is broken

### ISSUE THREAD

**Issue by DHenryDai** [getinsomnia/insomnia#3484]
**Describe the bug**
Switching Content Type "text/othe"r on Insomnia POST requests has no effect on Content Header.

**To Reproduce**
* Create a POST Request in Insomnia and select the Content Type (e.g. text/edn)
* Switch the Content Type to text/other
* Submit your request
* check Request timeline
* Content Type stays "text/edn"

![grafik](https://user-images.githubusercontent.com/86047904/122365742-39956e00-cf5b-11eb-8905-11aadf914aef.png)

**Expected behavior**
My expectation is that when i explicitely select the content header, there is a modification on submitted content header.
Option 1) clear the content header field in request
Option 2) because the option "other" is grouped under "text": set text/plain as Content-Header field
![grafik](https://user-images.githubusercontent.com/86047904/122365072-a6f4cf00-cf5a-11eb-9902-e125ed90714d.png)

I prefer Option 2

**Screenshots**
see above

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Installation Method: Traditional Autoupdate
 - App Version 2021.03.0

**Additional context**
nothing

Thanks in advance

**Comment by theozordako** [comment]
Will fix this issue

**Comment by develohpanda** [comment]
Feels like option one would be better 🤔 If a user is selecting a custom content type, then Insomnia should clear anything it set automatically, that seems reasonable. We can add "text/plain" as a standalone option (as per https://github.com/Kong/insomnia/pull/3793), but it doesn't seem right for "other" to fallback to "plain"

**Comment by develohpanda** [comment]
According to the [spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec7.html#sec7.2.1), when the content-type is not known it should be treated as `application/octet-stream`, but since we already have a `Binary File` body type which defaults to `application/octet-stream`, I think we should clear the content-type header if "Other" is selected as a body type.

**Comment by DHenryDai** [comment]
Yes, i see the requirement here to remove already set content types in header. does it make sense to inform the user by a slide-in element that the header content type war reset?

**Comment by develohpanda** [comment]
@DHenryDai hmmm, I think we can get away without any alerts or toasts, behaving the same way as when you select "No Body"
